README.md

```diff
@@ -1,39 +1,362 @@
+### Using Dependency Injection
+
+Here is a simple code that does dependency injection using `get_it`.
+
+```dart
+   GetIt getIt = GetIt.instance;
+
+   // Register the ReviewPrompter implementation with GetIt
+   getIt.registerSingleton<ReviewPrompter>(ReviewPrompter());
+```
+
+### Adding your CustomReviewSettings
+
+First let's declare a new ReviewSettings named CustomReviewSettings with a favoriteJokesCount to track how many jokes were favorited in an hypothetical dad jokes application
+
+```dart 
+  class CustomReviewSettings extends ReviewSettings {
+    final int favoriteJokesCount;
+
+    const CustomReviewSettings({
+      this.favoriteJokesCount = 0,
+      super.primaryActionCompletedCount = 0,
+      super.secondaryActionCompletedCount = 0,
+      super.applicationLaunchCount = 0,
+      super.firstApplicationLaunch,
+      super.requestCount = 0,
+      super.lastRequest,
+    });
+
+    @override
+    CustomReviewSettings copyWith({
+      int? primaryActionCompletedCount,
+      int? secondaryActionCompletedCount,
+      int? applicationLaunchCount,
+      DateTime? firstApplicationLaunch,
+      int? requestCount,
+      DateTime? lastRequest,
+    }) {
+      return CustomReviewSettings(
+        primaryActionCompletedCount:
+            primaryActionCompletedCount ?? this.primaryActionCompletedCount,
+        secondaryActionCompletedCount:
+            secondaryActionCompletedCount ?? this.secondaryActionCompletedCount,
+        applicationLaunchCount:
+            applicationLaunchCount ?? this.applicationLaunchCount,
+        firstApplicationLaunch:
+            firstApplicationLaunch ?? this.firstApplicationLaunch,
+        requestCount: requestCount ?? this.requestCount,
+        lastRequest: lastRequest ?? this.lastRequest,
+        favoriteJokesCount: favoriteJokesCount,
+      );
+    }
+
+    CustomReviewSettings copyWithFavorite(int favoriteCount) {
+      return CustomReviewSettings(
+        primaryActionCompletedCount: super.primaryActionCompletedCount,
+        secondaryActionCompletedCount: super.secondaryActionCompletedCount,
+        applicationLaunchCount: super.applicationLaunchCount,
+        firstApplicationLaunch: super.firstApplicationLaunch,
+        requestCount: super.requestCount,
+        lastRequest: super.lastRequest,
+        favoriteJokesCount: favoriteCount,
+      );
+    }
+  }
+```
+
+Notes : 
+  1. It's important you define a copyWith that overrides the one from the ReviewSettings class while returning the newly added property. This ensures that the calls made to the copyWith method of ReviewSettings will be overriden by your implementation
+  2. You need to define a copyWith method that returns the parent reviewSettings value along with your new property passed as a parameter. This method will be used when you will track your review settings further down the line. 
+
+Once you've defined your new custom review settings you need to add an extension to the ReviewConditionsBuilder like this :
```

Comment:
This is not needed but rather recommended. This makes it so that the API surface is more uniform.

Reply:
Updated.